aplib.net-demo/Assets/Scripts/Inventory/Weapons/WeaponInventory.cs

```diff
@@ -0,0 +1,103 @@
+using Entities.Weapons;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class WeaponInventory : AbstractInventory
+{
+    public Queue<Weapon> weaponList;
+
+    public Weapon _meleeWeapon;
+    public Weapon _rangedWeapon;
+    public void Start()
+    {
+        _inventoryIndicator = GetComponent<RawImage>();
+
+        weaponList = new Queue<Weapon>();
+        //once we get another way to acquire weapons, this can be removed
+
+        PickUpWeapon(_meleeWeapon);
+        PickUpWeapon(_rangedWeapon);
+        DisplayItem();
+    }
+    /// <summary>
+    /// Picks up a weapon and puts it in the inventory.
+    /// </summary>
+    /// <param name="weapon"></param>
+    public override void PickUpWeapon(Weapon weapon)
+    {
+        bool alreadyInInventory = false;
+        List<Weapon> _tempWeaponList = weaponList.ToList();
+        for (int i = 0; i < weaponList.Count; i++)
+        {
+            if (_tempWeaponList[i].name == weapon.name)
+            {
+                if (!weapon.stackable)
+                    return;
+
+                _tempWeaponList[i].uses += weapon.startUses;
+                alreadyInInventory = true;
+                break;
+            }
+        }
+
+        if (!alreadyInInventory)
+        {
+            weaponList.Enqueue(weapon);
+            DisplayItem();
+        }
+    }/// <summary>
```

Comment:
Please look at Joachim's code in the inventory on how to do this more efficiently (The !item.stackable part should probably be moved down a little bit, I think it will now not allow you to store any non-stackable items at all)